src/btreemap.rs

```diff
@@ -509,38 +509,39 @@ where
         node.save(self.allocator_mut());
     }
 
-    /// Returns the value associated with the given key if it exists.
+    /// Returns the value for the given key, if it exists.
     pub fn get(&self, key: &K) -> Option<V> {
         if self.root_addr == NULL {
             return None;
         }
+        self.traverse(self.root_addr, key, |node, idx| {
+            node.into_entry(idx, self.memory()).1 // Extract value.
+        })
+        .map(Cow::Owned)
+        .map(V::from_bytes)
+    }
 
-        self.get_helper(self.root_addr, key)
-            .map(Cow::Owned)
-            .map(V::from_bytes)
+    /// Returns true if the key exists.
+    pub fn contains_key(&self, key: &K) -> bool {
+        // An empty closure returns Some(()) if the key is found.
+        self.root_addr != NULL && self.traverse(self.root_addr, key, |_, _| ()).is_some()
     }
 
-    fn get_helper(&self, node_addr: Address, key: &K) -> Option<Vec<u8>> {
+    /// Recursively traverses from `node_addr`, calling `f` on a match; stops at a leaf.
+    fn traverse<F, R>(&self, node_addr: Address, key: &K, f: F) -> Option<R>
+    where
+        F: Fn(Node<K>, usize) -> R + Clone,
+    {
         let node = self.load_node(node_addr);
         match node.search(key) {
-            Ok(idx) => Some(node.into_entry(idx, self.memory()).1),
-            Err(idx) => {
-                match node.node_type() {
-                    NodeType::Leaf => None, // Key not found.
-                    NodeType::Internal => {
-                        // The key isn't in the node. Look for the key in the child.
```

Comment:
Perhaps preserve this comment in the new code? Or you thought it's not particularly useful?

Reply:
Rearranged the code, added comments for each leg.